Add env command

jdmundrawala opened the pull request:
```
jmundrawala@seamundrawala01 ~/workspace/chef-dk (jdm/chef-env●●)$ bundle exec chef env
---
ChefDK Environment:
  ChefDK Home: "/Users/jmundrawala/.chefdk"
  Path:
  - "/opt/chefdk/"
  - "/Users/jmundrawala/.gem/ruby/2.1.0/bin"
  - "/opt/chefdk/"
  - "/Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/bin"
  - "/Users/jmundrawala/.rbenv/versions/2.1.2/bin"
  - "/usr/local/Cellar/rbenv/0.4.0/libexec"
  - "/Users/jmundrawala/.local/bin"
  - "/Users/jmundrawala/go/bin"
  - "/Users/jmundrawala/.rbenv/shims"
  - "/usr/local/bin"
  - "/usr/local/sbin"
  - "/usr/bin"
  - "/bin"
  - "/usr/sbin"
  - "/sbin"
  - "/opt/X11/bin"
  - "/Users/jmundrawala/.yadr/bin"
  - "/Users/jmundrawala/.yadr/bin/yadr"
  Gem Environment:
    GEM ROOT: "/Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0"
    GEM HOME: "/Users/jmundrawala/.gem/ruby/2.1.0"
    GEM PATHS:
    - "/Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0"
    - "/Users/jmundrawala/.gem/ruby/2.1.0"
  Policyfile Config:
    Cache Path: "/Users/jmundrawala/.chefdk/cache"
    Storage Path: "/Users/jmundrawala/.chefdk/cache/cookbooks"
```
cc @danielsdeleo

danielsdeleo commented:
This is pretty rad. Only thing maybe to add would be versions of ruby, gems, and chef-dk, possibly some platform data, like in `gem env`:

```yaml
  - RUBYGEMS VERSION: 2.4.6
  - RUBY VERSION: 2.1.5 (2014-11-13 patchlevel 273) [x86_64-darwin14.0]
# snip..
  - RUBYGEMS PLATFORMS:
    - ruby
    - x86_64-darwin-14
```

Just trying to think if this might be more useful as a debug tool with that stuff (?)

danielsdeleo commented on lib/chef-dk/command/env.rb:
```diff
@@ -0,0 +1,61 @@
+          chefdk_env['Gem Environment'] = gem_environment
+          chefdk_env['Policyfile Config'] = policyfile_config
+        end
+        puts info.to_yaml
```
`ui` ?

danielsdeleo commented:
This looks good. Probably don't have to go overboard with testing, but it would be good to make sure it at least doesn't throw an exception when you run it.

jdmundrawala commented:
Added more info, did some reorganizing, and added a spec.
new output:
```yaml
---
Chef Development Kit:
  ChefDK:
    ChefDK Version: 0.6.0
    ChefDK Home: "/Users/jmundrawala/.chefdk"
    Policyfile Config:
      Cache Path: "/Users/jmundrawala/.chefdk/cache"
      Storage Path: "/Users/jmundrawala/.chefdk/cache/cookbooks"
  Ruby:
    Ruby Executable: "/Users/jmundrawala/.rbenv/versions/2.1.2/bin/ruby"
    Ruby Version: 2.1.2
    RubyGems:
      RubyGems Version: 2.4.4
      RubyGems Platforms:
      - ruby
      - x86_64-darwin-13
      Gem Environment:
        GEM ROOT: "/Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0"
        GEM HOME: "/Users/jmundrawala/.gem/ruby/2.1.0"
        GEM PATHS:
        - "/Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0"
        - "/Users/jmundrawala/.gem/ruby/2.1.0"
  Path:
  - "/opt/chefdk"
  - "/Users/jmundrawala/.gem/ruby/2.1.0/bin"
  - "/opt/chefdk"
  - "/Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/bin"
  - "/Users/jmundrawala/.rbenv/versions/2.1.2/bin"
  - "/usr/local/Cellar/rbenv/0.4.0/libexec"
  - "/Users/jmundrawala/.local/bin"
  - "/Users/jmundrawala/go/bin"
  - "/Users/jmundrawala/.rbenv/shims"
  - "/usr/local/bin"
  - "/usr/local/sbin"
  - "/usr/bin"
  - "/bin"
  - "/usr/sbin"
  - "/sbin"
  - "/opt/X11/bin"
  - "/Users/jmundrawala/.yadr/bin"
  - "/Users/jmundrawala/.yadr/bin/yadr"
```

danielsdeleo commented:
:100: :+1: :stars: :star2:

tyler-ball commented:
What about `ChefDK Install Directory`?  Useful?

tyler-ball commented on lib/chef-dk/command/env.rb:
```diff
@@ -0,0 +1,89 @@
+#
+# Copyright:: Copyright (c) 2014 Chef Software Inc.
```
(nit) 2015

tyler-ball commented on lib/chef-dk/command/env.rb:
```diff
@@ -0,0 +1,89 @@
+        info['Chef Development Kit'] = Hash.new.tap do |chefdk_env|
+          chefdk_env['ChefDK'] = chefdk_info
+          chefdk_env['Ruby'] = ruby_info
+          chefdk_env['Path'] = paths
```
(nit) `chefdk_env['Paths']`

danielsdeleo replied:
I'm cool with it being singular, because the actual env var is PATH

tyler-ball replied:
:+1:

tyler-ball commented:
![](http://i.imgur.com/45uA8.jpg)

jdmundrawala commented:
moving to https://github.com/chef/chef-dk/pull/423
